compiler/passes/scopeResolve.cpp

```diff
@@ -1436,14 +1442,14 @@ static void resolveModuleCall(CallExpr* call) {
 
                 call->replace(new UnresolvedSymExpr(mbrName));
 
-                parent->insertAtHead(mod);
+                parent->insertAtHead(modArg);
                 parent->insertAtHead(gModuleToken);
               }
 
             } else if (CallExpr* c = resolveModuleGetNewExpr(call, sym)) {
               call->replace(new SymExpr(sym));
 
-              c->insertAtHead(mod);
+              c->insertAtHead(modArg);
```

Comment:
This is the right call for functionality.  Are there situations where this could cause a confusing error message for the user (like, if we think we have a function match, but they actually called it with the wrong arguments?).  I think that's less likely for these qualified access cases (as I think we probably just talk about the function/method name and its arguments rather than the full statement that was provided), but that is something we've had to worry about in the past for things like renaming (which similarly modifies the AST to something that matches what we resolved to but doesn't look like what the user wrote any more)

Reply:
Do you have a (semi-)concrete example of this?

Reply:
```chapel
module A {
  proc foo(x: int) {
    writeln(x);
  }
}
module B {
  import A.foo;
}
module User {
  import B;

  proc main() {
    B.foo(); // Note: no arguments, but A.foo requires an int
  }
}
```

Reply:
After this last case, I don't feel the need to look at this again :)

Reply:
We get the correct error in this case right now. I added this as a test.